src/platform/site-wide/mega-menu/containers/Main.jsx

```diff
@@ -23,10 +22,56 @@ export function getAuthorizedLinkData(loggedIn, authenticatedLinks = authenticat
 }
 
 export class Main extends React.Component {
+
+  toggleDropDown = (currentDropdown) => {
+    const isVisible = !!currentDropdown;
+    if (isVisible) {
+      recordEvent({
+        event: 'nav-header-top-level',
+        'nav-header-action': `Navigation - Header - Open Top Level - ${currentDropdown}`
+      });
+    }
+    this.props.togglePanelOpen(currentDropdown);
```

Comment:
@cehsu @erikphansen Good example of where having a redux middleware handle analytics could come in handy.

Reply:
Yeah that's a great idea. I would love to see an analytics layer that could remove it from the rest of our code.